src/main/scala/org/allenai/ike/IkeBatchSearch.scala

```diff
@@ -0,0 +1,338 @@
+    // table with an 'Energy' column that has some seed entries like "light" and "heat" and
+    // we would like to find similar words like "thermal energy" and "electricity", the
+    // `tableExpander` will compute the word2vec centroid of the specified column's current
+    // entries and pull up n closest neighbors. The query that would trigger this will look
+    // like: `$MaterialConductivity.Energy ~ 10`.
+    val tableExpander = new SimilarPhrasesBasedTableExpander(word2vecPhrasesSearcher)
+
+    /** Search for a pattern with a pre-created @Searcher and save the results to the given file.
+      * Results are saved as JSON-serialized @BlackLabResult objects.
+      *
+      * @param searcher the searcher to query with
+      * @param pattern the pattern to search
+      * @param outputFile the file results should be saved in
+      */
+    def searchForPattern(searcher: Searcher, pattern: NamedPattern, outputFile: File): Unit = {
+      SearchApp.parse(pattern.pattern) match {
+        case Success(query) =>
+          // Get a simplified (ready-to-execute) version of the query.
+          val interpolatedQuery = QueryLanguage.interpolateQuery(
+            query,
+            tables,
+            patterns,
+            word2vecPhrasesSearcher,
+            tableExpander
+          ).get
+
+          val results = SearchApp.search(
+            interpolatedQuery,
+            searcher,
+            batchSearchOptions.textSource,
+            Some(WINDOW_SIZE))
+          Resource.using(new PrintWriter(outputFile)) { writer =>
+            results.foreach { result =>
+              writer.println(mapper.writeValueAsString(result))
+            }
+          }
+
+        case Failure(f) =>
+          logger.error(s"Unable to parse pattern  ${pattern.name}", f)
+      }
+    }
+
+    /** Split a pattern line into a pattern name and table name, as a tuple
+      *
+      * @param patternLine the pattern line to split
+      * @return a tuple with (pattern name, table name)
+      */
+    def splitPattern(patternLine: String): Option[(String, String)] = {
+      patternLine.split("\t").map(_.trim) match {
+        case Array(patternName, tableName, _*) => Some((patternName, tableName))
+        case _ => None
+      }
+    }
+
+    val sparkConf = new SparkConf()
+      .setAppName(this.getClass.getSimpleName)
+      .setMaster("local[*]")
+      .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .set("spark.kryo.registrator", "org.allenai.ike.IkeKryoRegistrator")
+    val sparkContext = new SparkContext(sparkConf)
+
+    // Open the index directory or create a temp directory if one hasn't been provided.
+    val indexParentDir = batchSearchOptions.indexPath.map { path =>
+      val dir = new File(path)
+      dir.mkdir()
+      dir
+    }.getOrElse {
+      val tempDir = Files.createTempDirectory("ike").toFile
+      tempDir.deleteOnExit()
```

Comment:
You can't `deleteOnExit` for directories. I wrote a workaround for that once, but it's part of the datastore code, so not in a super accessible part of the code.